tldw_Server_API/app/api/v1/endpoints/media/process_mediawiki.py

```
@@ -0,0 +1,241 @@
+
+router = APIRouter()
+
+
+@router.post(
+    "/mediawiki/ingest-dump",
+    summary="Ingest and process a MediaWiki XML dump, storing results to database and vector store.",
+    tags=["MediaWiki Processing"],
+    dependencies=[Depends(guard_backpressure_and_quota)],
+)
+async def ingest_mediawiki_dump_endpoint(
+    form_data: Dict[str, Any] = Depends(legacy_media.get_mediawiki_form_data),  # type: ignore[attr-defined]
```

Ruff (0.14.5)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)